moshi/src/main/java/com/squareup/moshi/LinkedHashTreeMap.java

```diff
@@ -346,13 +346,11 @@ private void rebalance(Node<K, V> unbalanced, boolean insert) {
         int rightLeftHeight = rightLeft != null ? rightLeft.height : 0;
 
         int rightDelta = rightLeftHeight - rightRightHeight;
-        if (rightDelta == -1 || (rightDelta == 0 && !insert)) {
-          rotateLeft(node); // AVL right right
-        } else {
+        if (rightDelta != -1 && (rightDelta != 0 || insert)) {
           assert (rightDelta == 1);
           rotateRight(right); // AVL right left
-          rotateLeft(node);
         }
+        rotateLeft(node); // AVL right right
```

Comment:
These comments no longer make sense. This `right right` comment is now being applied to the right-left case.